### PR TITLE
Markdown core: turndown/marked conversion, output/input, frontmatter, chunking, AI-Tools flags

### PR DESCRIPTION
**Pull request by msoukhomlinov**
## Summary

Adds Markdown as a first-class content format across the node, so Hudu can serve as a clean, citable knowledge source for AI/MCP workflows. Replaces the hand-rolled regex HTML→Markdown converter with maintained libraries and wires Markdown output/input, source frontmatter, and heading-chunking across Articles, Magic Dash, and Assets — plus the AI-Tools node.

This is the full self-hosted edition (`n8n-nodes-hudu`), which already ships runtime deps and is intentionally **not** n8n-Cloud-verifiable (the verified variant is the separate `n8n-nodes-hudu-cloud` package). Accordingly this branch switches lint to `configWithoutCloudSupport` (`n8n.strict=false`) so the Cloud-only scanner rules no longer apply to this edition.

## What's included

- **Conversion core** (`nodes/Hudu/utils/markdown/`): `turndown` + `turndown-plugin-gfm` (HTML→MD, GFM tables/strikethrough/task-lists), `marked@^4` (MD→HTML, pinned to v4 since v5+ are pure-ESM and break the CommonJS build), `frontmatter` (dependency-free, injection-safe YAML), `chunk` (fence-aware heading split). The public `convertHtmlToMarkdown(html)` signature is unchanged; the regex converter is removed.
- **Articles**: `Include Markdown Content` now backed by turndown; new `Include Source Frontmatter` (title/url/company/folder/updated_at); new `Chunk By Heading` (get/getAll) that fans one article into one output item per section with flat per-chunk source metadata for RAG/vector ingestion; new `Content Format: HTML | Markdown` on create/update.
- **Magic Dash**: Markdown output + frontmatter (title/company_name/content_link — no URL) and `Content Format` input.
- **Assets**: `Include Field Markdown` adds a Markdown sibling to RichText custom-field values (heuristic HTML detection).
- **AI-Tools node**: `output_markdown` + `include_frontmatter` (Articles) / `output_markdown` (Assets) read flags, and `content_format` write flag for Articles — which also fixes a latent bug where the create schema advertised "HTML or Markdown" but never converted. No `chunk` flag (a single tool call can't fan out items).

## Testing

New `vitest` harness (devDep only; tests excluded from the build and the published package). 19 tests: converter, MD→HTML round-trip, snapshot, frontmatter injection-escaping, chunk edge cases (fence-awareness, preamble, multi-heading), `looksLikeHtml`, and `applyContentFormat`. `npm run build` clean.

## Follow-ups (not in this PR)

- #37 — Assets Markdown **input** (MD→HTML on RichText create/update)
- #38 — register Magic Dash as an AI-tools resource before adding its MD flags

## Notes for review

- Behaviour is byte-identical between the regular node and the AI-Tools path (shared `processArticleContent` / `addAssetFieldMarkdown`).
- Markdown output is opt-in (all toggles default false).
- The `markdown_content` text differs slightly from the old regex converter for users who previously opted in — noted in CHANGELOG; opt-in, so no workflow change required.

### CODE REVIEW

File: nodes/Hudu/utils/markdownUtils.ts [hunk 1]
```diff
@@ -8,35 +11,63 @@ import { convertHtmlToMarkdown } from './htmlToMarkdown';
...
+    chunk_count: chunks.length,
+    heading: c.heading,
+    heading_level: c.level,
+    markdown: c.content,
+    source_type: 'article',
+    source_id: article.id ?? null,
+    title: article.name ?? null,
+    url: article.url ?? null,
+    company_id: article.company_id ?? null,
+    folder_id: article.folder_id ?? null,
```
[the remark this answers is not in the document]
Comment (msoukhomlinov) [reply nodes/Hudu/utils/markdownUtils.ts:86]: Addressed in e833072. `buildArticleChunks` now carries `company_id_label`, `folder_id_label`, `folder_path`, and `folder_description` onto every chunk when the handler has enriched the article (Include Company/Folder Details), instead of projecting only raw IDs. Added `buildArticleChunks.test.ts` covering the enrichment passthrough and the no-enrichment case. Full suite 22/22 green.

**Comment by msoukhomlinov** [comment]
@codex, review
